domain/domain.go

```diff
@@ -286,8 +296,7 @@ func (do *Domain) tryLoadSchemaDiffs(m *meta.Meta, usedVersion, newVersion int64
 			return nil, nil, err
 		}
 		if diff == nil {
-			// If diff is missing for any version between used and new version, we fall back to full reload.
-			return nil, nil, fmt.Errorf("failed to get schemadiff")
+			continue
```

Comment:
Please add a comment like

```
// Empty diff means the txn of generating schema version is committed, but the txn of `runDDLJob` is not. It is safe to skip the empty diff because the infoschema is new enough and consistent.
```